compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -904,6 +904,12 @@ public void visit(BLangLambdaFunction bLangLambdaFunction) {
         SymbolEnv symbolEnv = env.createClone();
         bLangLambdaFunction.capturedClosureEnv = symbolEnv;
         BLangFunction enclInvokable = (BLangFunction) symbolEnv.enclInvokable;
+        // Since lambda functions enclosed in a query lambda are not included in the
+        // package node lambda list need to visit them here.
+        if (enclInvokable.flagSet.contains(Flag.QUERY_LAMBDA)) {
```

Comment:
Any particular reason why we do not model anonymous functions generated for query expressions similar to normal anonymous functions?

I believe there's some improvements we can do for normal anonymous functions (like local anonymous functions not having to be available at module-level until desugar) but desugar and after, shouldn't have much of a difference, right?

Reply:
This was done to ensure that the lambda functions created for query clauses would be visited prior to the local anonymous functions in query clauses so that the closures would be resolved correctly. Updated the query desugar to handle it from the query side itself.